vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -1160,34 +1563,29 @@ private void reset() {
                 "if($0.$connector) $0.$connector.reset();", getElement()));
     }
 
-    private DataCommunicator<T> getDataCommunicator() {
-        initDataCommunicator();
-        return dataCommunicator;
-    }
-
-    private void initDataCommunicator() {
-        if (dataCommunicatorInitializer != null) {
-            /*
-             * Init the Data Communicator: 1. Lazily, when the data lazy loading
-             * is used. Initialization occurs when the user clicks on the
-             * dropdown to view the list of items. 2. Eagerly, when the items
-             * are set explicitly or in-memory Data Provider is used.
-             */
-            dataCommunicatorInitializer.init();
-        } else if (dataCommunicator == null) {
-            /*
-             * If the user hasn't provided any data, initialize with empty data
-             * set.
-             */
-            setItems();
+    private void initDataCommunicatorIfEmpty() {
+        if (dataCommunicator == null) {
+            setItems(new DataCommunicator.EmptyDataProvider<>());
         }
     }
 
-    /**
-     * Callback for Data Communicator lazy initialization
-     */
-    @FunctionalInterface
-    private interface DataCommunicatorInitializer extends Serializable {
-        void init();
+    private void enableDataProviderFetch() {
+        dataCommunicator.setFetchDisabled(false);
```

Comment:
Oh damn my bad review, just now realized that there is a double negation which is against our conventions 😞 
Should be `fetchEnabled` instead

Reply:
Thanks! Will change the method name

Reply:
Done